netbox/extras/migrations/0108_script_models.py

```diff
@@ -0,0 +1,92 @@
+# Generated by Django 4.2.9 on 2024-02-05 21:37
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def update_event_rules(apps, schema_editor):
+    from extras.models import ScriptModule
+    Script = apps.get_model('extras', 'Script')
+    ContentType = apps.get_model('contenttypes', 'ContentType')
+    ct = ContentType.objects.filter(app_label='extras', model='script').first()
+
+    EventRule = apps.get_model('extras', 'EventRule')
+    ct_script_module = ContentType.objects.filter(app_label='extras', model='scriptmodule').first()
+    for rule in EventRule.objects.filter(action_object_type=ct_script_module):
+        name = rule.action_parameters.get('script_name')
+        obj, created = Script.objects.get_or_create(
+            module_id=rule.action_object_id,
+            name=name,
+            defaults={'is_valid': False}
+        )
+
+        rule.action_object_type = ct
+        rule.action_object_id = obj.id
+        rule.save()
+
+
+def update_scripts(apps, schema_editor):
+    from extras.models import ScriptModule
```

Comment:
Importing models directly can cause issues with future releases as code changes over time. Unfortunately, the safest option here is to forklift any necessary logic (e.g. `get_module_scripts()`) into the migration itself to protect it from future changes to the model.